src/plot_api/plot_schema.js

```diff
@@ -9,185 +9,287 @@
 
 'use strict';
 
-var Plotly = require('../plotly');
 var Registry = require('../registry');
-var Plots = require('../plots/plots');
 var Lib = require('../lib');
 
-// FIXME polar attribute are not part of Plotly yet
+var baseAttributes = require('../plots/attributes');
+var baseLayoutAttributes = require('../plots/layout_attributes');
+var frameAttributes = require('../plots/frame_attributes');
+var animationAttributes = require('../plots/animation_attributes');
+
+// polar attributes are not part of the Registry yet
 var polarAreaAttrs = require('../plots/polar/area_attributes');
 var polarAxisAttrs = require('../plots/polar/axis_attributes');
 
 var extendFlat = Lib.extendFlat;
 var extendDeep = Lib.extendDeep;
-var extendDeepAll = Lib.extendDeepAll;
 
-var NESTED_MODULE = '_nestedModules',
-    COMPOSED_MODULE = '_composedModules';
+var IS_SUBPLOT_OBJ = '_isSubplotObj';
+var IS_LINKED_TO_ARRAY = '_isLinkedToArray';
+var DEPRECATED = '_deprecated';
+var UNDERSCORE_ATTRS = [IS_SUBPLOT_OBJ, IS_LINKED_TO_ARRAY, DEPRECATED];
```

Comment:
> _deprecated

These are simply deprecated keys that don't show up on https://plot.ly/javascript/reference/ and friends, but that plotly.py (still) considers valid.

> _isLinkedToArray

See https://github.com/plotly/plotly.js/pull/1144#discussion_r88044896

> _isSubplotObj

Subplot layout containers e.g. `layout.xaxis` , `layout.scene` can be input as `layout.xaxis2` or `layout.scene10`, `Plotly.validate` and plotly.py need to know about that.

> `role`

Has no effect in plotly.js (yet). At the moment, it is only used for plotly.py's strip style method which removes all `role: 'style'` keys from a given figure.